pkg/policy/multi.go

```diff
@@ -187,6 +187,9 @@ func (p *multiDenyProcessor) ProcessResource(val ast.Value) error {
 		p.builders[correlation].severity = p.metadata.Severity
 		p.builders[correlation].remediation = p.defaultRemediation
 	}
+	if result.Context != nil {
+		p.builders[correlation].context = result.Context
+	}
```

Comment:
I'm a bit wary of this, it probably won't cause any errors, but it goes a bit against the grain of Rego queries being incremental and idempotent.

If there are two resources rules (or, maybe more likely, a case where both `deny` and `resources` yield context), we would not want one rule be able to override the results of the other (since we can't make much guarantees about the Rego evaluation order).

I would suggest adding at least something along the lines of this to `ruleresultbuilder.go`:

```go
func (builder *ruleResultBuilder) addContext(context map[string]interface{}) {
	if len(context) == 0 {
		return
	}
	if builder.context == nil {
		builder.context = map[string]interface{}{}
	}
	for k, v := range context {
		builder.context[k] = v
	}
}
```

And then calling `addContext` here (and other places).

Reply:
Only `resources` will add context, at least for now. Since it is called in both pass/fail situations, I think it should be enough to provide the context in both situations. But agree, being idempotent in this case is more future proof.